xonsh/procs/readers.py

```diff
@@ -54,31 +54,27 @@ def read_queue(self):
 
     def read(self, size=-1):
         """Reads bytes from the file."""
-        i = 0
-        buf = b""
-        while size < 0 or i != size:
+        buf = bytearray()
+        while size < 0 or len(buf) != size:
             line = self.read_queue()
             if line:
                 buf += line
             else:
                 break
-            i += len(line)
         return buf
 
     def readline(self, size=-1):
         """Reads a line, or a partial line from the file descriptor."""
-        i = 0
         nl = b"\n"
-        buf = b""
-        while size < 0 or i != size:
+        buf = bytearray()
+        while size < 0 or len(buf) != size:
```

Comment:
I wonder if this shouldn't be `len(buf) <= size` instead and have the method return a truncated value.